spec/src/main/asciidoc/repository.asciidoc

```diff
@@ -170,31 +182,86 @@ public interface OrderRepository extends CrudRepository<Order, Long> {
 }
 ----
 
-For queries by method name, the resolution algorithm starts by interpreting the whole part (AddressZipCode) as the property and checks the domain class for a property with that name (uncapitalized). If the algorithm succeeds, it uses that property.
+The resolution algorithm for identifying properties in query methods by method name, with manual traversal points, is defined as follows:
+
+1. *Method Name Parsing*:: The query method's name is parsed to identify the property or properties being referenced. Method names in query methods typically follow a pattern of "findBy[Property]", where "[Property]" represents the name of the property you want to query by.
+
+2. *Property Extraction*:: The property name is extracted from the method name by removing the "findBy" prefix. For example, in the query method `findByAddressZipCode`, the property name extracted is `AddressZipCode`.
+
+3. *Property Name Capitalization*:: The extracted property name is treated as is, with its original capitalization. For example, if the property name is `AddressZipCode`, it remains in camel case.
```

Comment:
IMHO: Nope

Reply:
What do you think?

Reply:
Personally, I'd be in favor of case insensitive matching for property names unless the vendors have an issue with it.  Using the example from this pull, a lot of users who define an entity attribute called `zipcode` won't have a clear understanding of whether that means its name is "zipcode" or "Zipcode" and a lot of them will probably intuitively write method names like findByZipCode that doesn't match the case of either.  Being case insensitive would be more flexible and spare the user from having to experience errors in these cases.  It seems unfortunate to complain about differences like that.  Maybe the case insensitive parsing (or case conversion) makes it more expensive though.

Reply:
This expensive part makes me afraid; that is why I suggest keeping the same following what we have with Spring and Micronout.

Reply:
I wouldn't just assume it's a problem.  Vendors like Micronaut would likely being doing their processing of methodNames at compile time.  Even a vendor that processes at run time probably only does so once and reuses.  It might not be much of a problem at all.